benchmark_runner/common/ocp_resources/create_ocp_resource_operations.py

```diff
@@ -102,7 +102,8 @@ def apply_non_approved_patch(self, approved_values_list: list, namespace: str, r
                 result = self.__oc.run(cmd=check_status).split()[index]
                 if result == 'true':
                     if resource == 'odf':
-                        for ind in range(3):
+                        num_status = self.__oc.run("oc get csv -n openshift-storage -ojsonpath={$.items[*].status.phase}")
+                        for ind in range(len(num_status.split())):
```

Comment:
The intent is to ensure that all CSVs in openshift-storage have succeeded?  If so, please update the title accordingly.

But you really want metadata.name rather than status.phase at line 105

Reply:
We need the number of CSV status that should be checked, actually no difference between status and name both are working

Reply:
It would be better to use the name, and then

```
verify_cmd=f"oc get csv -n {namespace} {name} -ojsonpath='{{.items[ind].status.phase}}'"
```

Then if the order of the CSVs in the list changes for some reason you're still OK.

Reply:
You're relying on the order of the returned CSVs to be the same from run to run.  Suppose it changes?

Reply:
No

Reply:
When I say "from run to run", I mean suppose the order returned is random?  You're assuming that index 1, for example, is going to be the same value each time.

Reply:
No what?

Reply:
You right, fixed it.